src/dynatrace/opentelemetry/metrics/export/serializer.py

```diff
@@ -133,44 +133,104 @@ def _get_metric_key(
     def _normalize_metric_key(key: str) -> str:
         first, *rest = key.split(".")
 
-        first = DynatraceMetricsSerializer._normalize_metric_key_first_section(
-            first
-        )
+        first = (DynatraceMetricsSerializer.
+                 __normalize_metric_key_first_section(first))
 
         if first == "":
             return ""
 
         rest = list(filter(None, map(
-            DynatraceMetricsSerializer._normalize_metric_key_section,
+            DynatraceMetricsSerializer.__normalize_metric_key_section,
             rest,
         )))
 
         return ".".join([x for x in [first] + rest if x != ""])
 
-    @staticmethod
-    def _normalize_metric_key_first_section(section: str) -> str:
-        return DynatraceMetricsSerializer._normalize_metric_key_section(
-            re.sub("^[^a-zA-Z]+", "", section),
+    # characters not valid to start the first identifier key section
+    __re_metric_key_first_identifier_section_start = (
+        re.compile(r"^[^a-zA-Z_]+"))
+
+    # characters not valid to start subsequent identifier key sections
+    __re_metric_key_identifier_section_start = re.compile(r"^[^a-zA-Z0-9_]+")
+
+    # for the rest of the metric key characters, alphanumeric characters as
+    # well as hyphens and underscores are allowed. consecutive invalid
+    # characters will be condensed into one underscore.
+    __re_metric_key_invalid_characters = re.compile(r"[^a-zA-Z0-9_\-]+")
+
+    @classmethod
+    def __normalize_metric_key_first_section(cls, section: str) -> str:
```

Comment:
Looks like `self` is used in other places. We should be consistent and use either `cls` or `self`

Reply:
we use `self` for instance methods and `cls` for class methods. PEP8 also suggests that, and i think it reads more easily. (<https://www.python.org/dev/peps/pep-0008/#function-and-method-arguments>)

Reply:
sounds good to me